client/src/components/relatedItems/RelatedItemsWidget.jsx

```diff
@@ -37,25 +43,72 @@ class RelatedItemsWidget extends Component {
       relatedProductsIds: [],
       loaded: false
     })
+
     this.setOverviewIdData();
   }
 
-
   setOverviewIdData() {
-
     this.setState({
       loaded: false
     })
-
     const endpoints = [
-      `http://localhost:8080/products/${this.state.overviewId}`,
-      `http://localhost:8080/products/${this.state.overviewId}/related`];
-
+      `/products/${this.state.overviewId}`,
+      `/products/${this.state.overviewId}/related`];
 
-    axios.all(endpoints.map((endpoint) => axios.get(endpoint)))
+    return axios.all(endpoints.map((endpoint) => axios.get(endpoint)))
       .then(axios.spread((overview, related) => {
 
         var overviewResult = overview.data;
+        var itemFeatures = overviewResult.features.map((x) => {
+          if (x.value !== null) {
+            return x.value;
+          }
+        });
+        var relatedResult = related.data;
+        var uniqueResults = [...new Set(relatedResult)].filter(id => id !== this.state.overviewId);
+        this.setState({
+          overviewIdName: overviewResult.name,
+          overviewIdFeatures: itemFeatures,
+          relatedProductsIds: uniqueResults,
+          loaded: true
+        });
+        return uniqueResults;
+      }))
+      .catch((err) => {
+        console.log('error in setOverviewIdData');
+        return err;
+      })
+  }
+
+  setArray(itemArray) {
+    // for each item in the array
+    var newArray = [];
+
+    itemArray.map((item) => {
+      newArray.push(setCards(item))
+    })
+
+    return newArray;
+  }
+
+  setCards(eachItem) {
+
+    const endpoints = [
+      `http://localhost:8080/styles/${eachItem}`,
```

Comment:
Change this localhost url as well then you will be good.

Reply:
Thank you!  I just pushed the new update with this localhost edited out.